src/Client/Updater.php

```diff
@@ -644,6 +643,79 @@ public function download(string $target, ...$extra): PromiseInterface
             });
     }
 
+    /**
+     * Gets a target metadata object that contains the specified target.
+     *
+     * @param string $target
+     *   The path of the target file. Needs to be known to the most recent
+     *   targets metadata downloaded in ::refresh().
+     * @param \Tuf\Metadata\TargetsMetadata|null $targetsMetadata
+     *   The targets metadata to search or null. If null then the search will
+     *   start at the top level 'targets.json' file.
+     * @param string[] $searchedRoles
+     *   The roles that have already been searched. This is used for recursive
+     *   calls to this function and should be provided by any callers.
+     *
+     * @return \Tuf\Metadata\TargetsMetadata|null
+     *   The target metadata with a match for the target, or null no match is
+     *   found.
+     */
+    protected function getMetadataForTarget(string $target, ?TargetsMetadata $targetsMetadata = null, array $searchedRoles = []): ?TargetsMetadata
+    {
+        if ($targetsMetadata === null) {
+            if (!empty($searchedRoles)) {
+                throw new \UnexpectedValueException('$searchedRoles should never be provided by outside calls to \Tuf\Client\Updater::getMetadataForTarget(). It is only used for recursive calls.');
```

Comment:
This feels aggressive. Why not simply empty the array out?

```suggestion
                $searchedRoles = [];
```

Reply:
Because the only reason this would ever happen is if some caller set this because they didn't understand the comment and how this is suppose to work. I don't want to swallow this error because it really means the caller does not understand how this suppose to work.

The should find out this problem rather have it be ignored because their misunderstanding could lead to other problems.